### PR TITLE
Condition ui view

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
Implemented:
1. Create Condition Set - adds to tree and opens in edit mode
2. Condition Edit (note: telemetry must be added to enable)
3. Add, remove, rename condition.
4. Change output to true or false or string value.
5. New condition displays selectors for any/all criteria are met, select telemetry, field, comparison with all options populated. Comparison of anything except id defined or is undefined shows value input.

Not implemented:
1. Drag conditions to re-order
2. Clone condition.
3. Save criteria values.
4. Add criteria.

### CODE REVIEW

File: src/plugins/condition/components/Condition.vue [hunk 1]
```diff
@@ -1,34 +1,20 @@
 <template>
-<div id="conditionArea"
+<div v-if="condition"
+     id="conditionArea"
      class="c-cs-ui__conditions"
-     :class="['widget-condition', { 'widget-condition--current': isCurrent }]"
+     :class="['widget-condition', { 'widget-condition--current': currentConditionIdentifier && (currentConditionIdentifier.key === conditionIdentifier.key) }]"
```
Comment (shefalijoshi) [inline src/plugins/condition/components/Condition.vue]: We should change this to use `openmct.objects.makeKeyString(<identifier>)` eventually.

File: src/plugins/condition/components/ConditionEdit.vue [hunk 2]
```diff
@@ -1,41 +1,27 @@
 <template>
-<div id="conditionArea"
+<div v-if="condition"
      class="c-cs-editui__conditions"
-     :class="['widget-condition', { 'widget-condition--current': isCurrent }]"
+     :class="['widget-condition', { 'widget-condition--current': currentConditionIdentifier && (currentConditionIdentifier.key === conditionIdentifier.key) }]"
```
Comment (shefalijoshi) [inline src/plugins/condition/components/ConditionEdit.vue]: We should change this to use `openmct.objects.makeKeyString(<identifier>) `at some point.

File: src/plugins/condition/components/ConditionEdit.vue [hunk 3]
```diff
@@ -1,41 +1,27 @@
...
-            <span v-if="isDefault"
-                  class="condition-description"
-            >When all else fails
-            </span>
-            <span v-else
-                  class="condition-description"
-            >[condition description]
-            </span>
+            <span class="condition-name">{{ condition.definition.name }}</span>
+            <span class="condition-description">{{ condition.definition.name }}</span>
```
Comment (shefalijoshi) [inline src/plugins/condition/components/ConditionEdit.vue]: condition description is displaying `condition.definition.name` - probably should be something else?

File: src/plugins/condition/utils/eventbus.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+import Vue from 'vue';
+export const EventBus = new Vue();
```
Comment (shefalijoshi) [inline src/plugins/condition/utils/eventbus.js]: Where is this being used?

File: src/plugins/condition/components/ConditionSet.vue [hunk 5]
```diff
@@ -2,9 +2,11 @@
 <div class="c-object-view u-contents">
```
Comment (shefalijoshi) [inline src/plugins/condition/components/ConditionSet.vue]: Please remove this outer div - `<div class="c-object-view u-contents">`

File: src/plugins/condition/ConditionSetViewProvider.js [hunk 6]
```diff
@@ -58,7 +58,7 @@ export default class ConditionSetViewProvider {
                             isEditing
                         }
                     },
-                    template: '<condition-set :isEditing="isEditing"></condition-set>'
+                    template: '<condition-set ref="conditionSet" :isEditing="isEditing"></condition-set>'
```
[the remark this answers is not in the document]
Comment (JoelMcKinnon) [reply src/plugins/condition/ConditionSetViewProvider.js]: It's not. Removed.

File: src/plugins/condition/components/Condition.vue (hunk 1 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue]: Added TODO

File: src/plugins/condition/components/ConditionEdit.vue (hunk 2 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/ConditionEdit.vue]: Added TODO

File: src/plugins/condition/components/ConditionEdit.vue (hunk 3 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/ConditionEdit.vue]: Added TODO: description should be derived from criteria

File: src/plugins/condition/utils/eventbus.js (hunk 4 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/utils/eventbus.js]: It's not. I removed it.

File: src/plugins/condition/components/ConditionSet.vue (hunk 5 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/ConditionSet.vue]: Done.

**Comment by shefalijoshi** [comment]
This PR is a bit unstable due to an erroneous commit. Opening a new PR to continue this merge. The code review is the same.

**Pull request closed without merging**